src/rust/ide/view/graph-editor/src/component/node/output/port.rs

```diff
@@ -445,16 +492,17 @@ impl Model {
                 |usage_tp,def_tp| usage_tp.clone().or_else(|| def_tp.clone())
             );
 
+            label.set_content <+ frp.tp.map(f!([](tp)
+                if let Some(tp) = tp {
+                    tp.to_string()
+                } else {
+                    "".to_string()
+                }
+            ));
```

Comment:
I think you're looking for this
`label.set_content <+ frp.tp.map(|tp| tp.map_or_else(||default(),|s| s.to_string()))`

Reply:
Btw, please add a function `map_or_default` to `Option` to our Prelude, and then use it like 
`label.set_content <+ frp.tp.map(|tp| tp.map_or_defaukt(|s| s.to_string()))`

This will be insanely useful function.

Reply:
I have created a [PR for the prelude](https://github.com/enso-org/rust-lib/pull/35). Who should review that?